fluent-react/src/localized.js

```diff
@@ -101,27 +101,36 @@ export default class Localized extends Component {
 
     const msg = bundle.getMessage(id);
     const [args, elems] = toArguments(this.props);
-    const messageValue = bundle.format(msg, args);
-
-    // Check if the fallback is a valid element -- if not then it's not
-    // markup (e.g. nothing or a fallback string) so just use the
-    // formatted message value
-    if (!isValidElement(elem)) {
-      return messageValue;
-    }
 
     // The default is to forbid all message attributes. If the attrs prop exists
     // on the Localized instance, only set message attributes which have been
     // explicitly allowed by the developer.
-    if (attrs && msg.attrs) {
+    if (attrs && msg.attributes) {
       var localizedProps = {};
       for (const [name, allowed] of Object.entries(attrs)) {
-        if (allowed && msg.attrs.hasOwnProperty(name)) {
-          localizedProps[name] = bundle.format(msg.attrs[name], args);
+        if (allowed && name in msg.attributes) {
+          localizedProps[name] = bundle.formatPattern(
+            msg.attributes[name], args);
         }
       }
     }
 
+    // If the message has a null value, we're only interested in its attributes.
+    // Do not pass the null value to cloneElement as it would nuke all children
+    // of the wrapped component.
+    if (msg.value === null) {
+      return cloneElement(elem, localizedProps);
+    }
+
+    const messageValue = bundle.formatPattern(msg.value, args);
+
+    // Check if the fallback is a valid element -- if not then it's not
+    // markup (e.g. nothing or a fallback string) so just use the
+    // formatted message value
+    if (!isValidElement(elem)) {
```

Comment:
The check to `isValidElement` moved to a weird spot, aka, you call `cloneElement` before this now?

Reply:
Thanks, I'll investigate.

Reply:
Nice catch, @Pike. I'm adding two unit tests to cover scenarios where this would have bitten us. Thanks again.